x/mongo/driver/operation.go

```diff
@@ -759,13 +759,7 @@ func (op Operation) Execute(ctx context.Context) error {
 		if ctx.Err() != nil {
 			err = ctx.Err()
 		} else if deadline, ok := ctx.Deadline(); ok {
-			if csot.IsTimeoutContext(ctx) && time.Now().Add(srvr.RTTMonitor().P90()).After(deadline) {
-				err = fmt.Errorf(
-					"remaining time %v until context deadline is less than 90th percentile RTT: %w\n%v",
-					time.Until(deadline),
-					ErrDeadlineWouldBeExceeded,
-					srvr.RTTMonitor().Stats())
-			} else if time.Now().Add(srvr.RTTMonitor().Min()).After(deadline) {
+			if time.Now().Add(srvr.RTTMonitor().Min()).After(deadline) {
 				err = context.DeadlineExceeded
```

Comment:
We should use `ErrDeadlineWouldBeExceeded` here with the additional info of the remaining time and the RTT stats (probably just the full list of 10 RTT samples).